go/sqltypes/bindvars.go

```diff
@@ -0,0 +1,279 @@
+/*
+Copyright 2017 Google Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package sqltypes
+
+import (
+	"errors"
+	"fmt"
+	"strconv"
+
+	"github.com/golang/protobuf/proto"
+
+	querypb "github.com/youtube/vitess/go/vt/proto/query"
+)
+
+// NullBV is a bindvar with NULL value.
+var NullBV = &querypb.BindVariable{Type: querypb.Type_NULL_TYPE}
+
+// BuildBindVars builds a map[string]*querypb.BindVariable from a map[string]interface{}.
```

Comment:
I hate to say that, but I don't like the BindVar name. It should be BindVariable:
- This file should still be bind_variables.go.
- NullBV should be NullBindVariable.
- BuildBindVars should be BuildBindVariables.
- Int64BindVariable.
and so on...